internal/pipeline/executor.go

```diff
@@ -303,19 +302,25 @@ func hydrate(p *pkg, hctx *hydrationContext) (resources []*yaml.RNode, err error
 	return resources, err
 }
 
-// filterMetaData filters kpt metadata files such as pipeline,
+// filterMetaResources filters kpt metadata files such as pipeline,
 // Kptfile, permissions etc.
-func filterMetaData(resources []*yaml.RNode) []*yaml.RNode {
+func filterMetaResources(resources []*yaml.RNode) []*yaml.RNode {
 	var filtered []*yaml.RNode
 	for _, r := range resources {
-		meta, _ := r.GetMeta()
-		if !strings.Contains(meta.APIVersion, "kpt.dev") {
+		m, _ := r.GetMeta()
+		if !strings.Contains(m.APIVersion, "kpt.dev") && !isFnConfig(&m) {
 			filtered = append(filtered, r)
 		}
 	}
 	return filtered
 }
 
+// isFnConfig filters out function configs
+func isFnConfig(m *yaml.ResourceMeta) bool {
+	_, hasFnAnnotation := m.Annotations["config.kubernetes.io/function"]
+	return m.Kind == "ConfigMap" && hasFnAnnotation
```

Comment:
Are you assuming that function configs MUST of type ConfigMap and MUST have such an annotation?

In general, a function config can be of any kind. And we shouldn't need this annotation anymore given we have an explicit Pipeline declaration. Function configs can be found by following the `config` field in the Pipeline.

@droot 

We still need a mechanism for such files to not be applied when calling `kpt live apply`. Current mechanism is to use `config.kubernetes.io/local-config: "true"` annotation.

Reply:
Before getting too deep into implementation, we need to have examples packaging using the v1alpha2 checked into `package-examples`. Otherwise, there's chance of misunderstanding the design spec.

Reply:
#1421

Reply:
understood, thank you for the input. will discuss with Sunil